crates/phactory/src/prpc_service.rs

```diff
@@ -1274,20 +1285,20 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> PhactoryApi for Rpc
         phactory.handover_ecdh_key = Some(handover_ecdh_key);
 
         let challenge = request.decode_challenge().map_err(from_display)?;
-        let attestation_provider = phactory.attestation_provider;
-
         let dev_mode = challenge.dev_mode;
-        let in_sgx = attestation_provider == Some(AttestationProvider::Ias);
-
         // generate local attestation report to ensure the handover pRuntimes are on the same machine
-        let sgx_local_report = if dev_mode || !in_sgx {
-            vec![]
-        } else {
-            let its_target_info =
-                unsafe { sgx_api_lite::decode(&challenge.sgx_target_info).unwrap() };
+        let sgx_local_report = if !dev_mode {
```

Comment:
What will happen if it's not in sgx mode?

Reply:
The receiver pRuntime will panic.